eth-contracts/test/serviceProvider.test.js

```diff
@@ -623,5 +623,28 @@ contract('ServiceProvider test', async (accounts) => {
         newSPId)
       assert.isTrue(newIdFound, 'Expected valid new ID')
     })
+
+    it('will modify the dns endpoint for an existing service', async () => {
+      const spId = await serviceProviderFactory.getServiceProviderIdFromEndpoint(testEndpoint)
+      const { endpoint } = await serviceProviderFactory.getServiceProviderInfo(testDiscProvType, spId)
+      assert.equal(testEndpoint, endpoint)
+      
+      // update the endpoint from testEndpoint to testEndpoint1
+      await serviceProviderFactory.updateEndpoint(testDiscProvType, testEndpoint, testEndpoint1, { from: stakerAccount })
+      const { endpoint: endpointAfter } = await serviceProviderFactory.getServiceProviderInfo(testDiscProvType, spId)
+      assert.equal(testEndpoint1, endpointAfter)
+
+      // it should replace the service provider in place so spId should be consistent
+      const spIdNew = await serviceProviderFactory.getServiceProviderIdFromEndpoint(testEndpoint1)
+      assert.isTrue(spId.eq(spIdNew))
+    })
+
+    it('will fail to modify the dns endpoint for the wrong owner', async () => {
```

Comment:
probably want a test case if you try to update an invalid endpoint

Reply:
@hareeshnagaraj how would you verify if it was invalid?

Reply:
confirm it fails when you try to update a non-existent endpoint

Reply:
using the assertRevert style logic

Reply:
good test case. done